File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        if(l1 == null && l2 == null)
            return null;
        else if(l1 != null && l2 == null)
            return l1;
        else if(l1 == null && l2 != null)
            return l2;
        
        ListNode nodes = new ListNode();
        ListNode result = nodes;
        
        while(l1 != null || l2 != null)
        {
            if(l1 != null && l2 != null)
            {
                int firstVal = l1.val;
                int secondVal = l2.val;
                
                if(firstVal < secondVal)
                {
                    nodes.next = new ListNode(firstVal);
                    l1 = l1.next;
                    nodes = nodes.next;
                }else{
                    nodes.next = new ListNode(secondVal);
                    l2 = l2.next;
                    nodes = nodes.next; 
                }
            }else if(l1 == null)
            {
                nodes.next = l2;
                return result.next;
            } else if(l2 == null)
            {
                nodes.next = l1;
                return result.next;
            }
        }
        
        return result.next;
        
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        left, right = list1, list2
        dummy = temp = ListNode()
        while(left or right):
            if(left is None):
                dummy.next = right
                break
            
            if(right is None):
                dummy.next = left
                break
                
            if(left.val < right.val):
                dummy.next = left
                left = left.next
            else:
                dummy.next = right
                right = right.next
            dummy = dummy.next



        return temp.next

